# Import PANDAS Library

Notice that we are using pd as an abbreviation so each time we want to use the PANDAS library, we can simply type pd instead of pandas.

In [1]:
import pandas as pd
pd.options.display.float_format = '${:,.2f}'.format

# Import Dataset (either full dataset or simplified)

You can use either set of datafiles (full or simplified). In the past some students' computers could not handle the full dataset. If you worry that your computer will not handle the full dataset (or if you find your code is running very slow), use the simplified version. 

The simplified version was created by taking a sample of the full data; hence the simplified files have "Sample" added to the end of the file names. Some files were not large to begin with and didn't need to be reduced. Those files do not have Sample added to the end of the file name.

## Adjust the file address below to match your computer.

You will want to replace the begining of the address below (i.e., everything before OneDrive - Oregon State University/)

In [2]:
begInv = pd.read_csv(r"C:\Users\msand\Oregon State University\Pesch, Heather Lynn - Bibitor_Data\2016Data\SimplifiedData\HW4 Sampled Data\BegInvFINAL12312016Sample.csv")

In [3]:
begInv.head(2)

InventoryId  Store          City  Brand                  Description  \
0  1_HARDERSFIELD_58      1  HARDERSFIELD     58  Gekkeikan Black & Gold Sake   
1  1_HARDERSFIELD_60      1  HARDERSFIELD     60       Canadian Club 1858 VAP   

    Size  onHand  Price   startDate  
0  750mL       8 $12.99  2016-01-01  
1  750mL       7 $10.99  2016-01-01

# Deliverable 1a

In [4]:
begInv = begInv.rename(columns = {'onHand':'01_01_2016_Qty'})
begInv.head(2)

InventoryId  Store          City  Brand                  Description  \
0  1_HARDERSFIELD_58      1  HARDERSFIELD     58  Gekkeikan Black & Gold Sake   
1  1_HARDERSFIELD_60      1  HARDERSFIELD     60       Canadian Club 1858 VAP   

    Size  01_01_2016_Qty  Price   startDate  
0  750mL               8 $12.99  2016-01-01  
1  750mL               7 $10.99  2016-01-01

In [5]:
begInv['InvCost01_01_2016'] = begInv['01_01_2016_Qty'] * begInv['Price']
begInv.head(2)

InventoryId  Store          City  Brand                  Description  \
0  1_HARDERSFIELD_58      1  HARDERSFIELD     58  Gekkeikan Black & Gold Sake   
1  1_HARDERSFIELD_60      1  HARDERSFIELD     60       Canadian Club 1858 VAP   

    Size  01_01_2016_Qty  Price   startDate  InvCost01_01_2016  
0  750mL               8 $12.99  2016-01-01            $103.92  
1  750mL               7 $10.99  2016-01-01             $76.93

In [6]:
begInv_Store = begInv[['Store','01_01_2016_Qty','InvCost01_01_2016']].groupby('Store').sum()
begInv_Store.head(2)

01_01_2016_Qty  InvCost01_01_2016
Store                                   
1               49917        $838,079.59
2               52925        $840,452.20

### One way to get this is to sort the in descending order and use head(10) to show the ten highest.

In [7]:
begInv_Store.sort_values(by='InvCost01_01_2016',ascending=False).head(10)

01_01_2016_Qty  InvCost01_01_2016
Store                                   
34             153852      $3,291,170.24
73             162551      $3,142,497.36
67             158996      $3,079,578.63
66             149314      $2,973,033.90
76             140208      $2,952,418.44
69             144255      $2,946,726.65
38             114368      $2,232,698.77
55             119641      $2,001,263.66
50              94720      $1,649,808.22
79              95330      $1,503,149.48

### Alternative way to get the same answer.

In [8]:
begInv_Store.sort_values(by='InvCost01_01_2016').tail(10).sort_values(by='InvCost01_01_2016', ascending = False)

01_01_2016_Qty  InvCost01_01_2016
Store                                   
34             153852      $3,291,170.24
73             162551      $3,142,497.36
67             158996      $3,079,578.63
66             149314      $2,973,033.90
76             140208      $2,952,418.44
69             144255      $2,946,726.65
38             114368      $2,232,698.77
55             119641      $2,001,263.66
50              94720      $1,649,808.22
79              95330      $1,503,149.48

# Deliverable 1b

In [9]:
begInv_Brand = begInv[['Brand','01_01_2016_Qty','InvCost01_01_2016']].groupby('Brand').sum()
begInv_Brand.head(2)

01_01_2016_Qty  InvCost01_01_2016
Brand                                   
58                281          $3,650.19
60                288          $3,165.12

In [10]:
begInv_Brand.sort_values(by='InvCost01_01_2016',ascending=False).head(10)

01_01_2016_Qty  InvCost01_01_2016
Brand                                   
3545            14499        $463,823.01
1233            12016        $432,455.84
8068            15341        $383,371.59
4261            15499        $340,823.01
3858            13649        $327,439.51
2753             4625        $286,703.75
8082             9287        $278,517.13
8680             7066        $275,503.34
2589             6766        $270,572.34
3876            14829        $266,773.71

# Deliverable 2a

In [11]:
endInv = pd.read_csv(r"C:\Users\msand\Oregon State University\Pesch, Heather Lynn - Bibitor_Data\2016Data\SimplifiedData\HW4 Sampled Data\EndInvFINAL12312016.csv")
endInv.head(2)

InventoryId  Store          City  Brand                  Description  \
0  1_HARDERSFIELD_58      1  HARDERSFIELD     58  Gekkeikan Black & Gold Sake   
1  1_HARDERSFIELD_62      1  HARDERSFIELD     62     Herradura Silver Tequila   

    Size  onHand  Price     endDate  
0  750mL      11 $12.99  2016-12-31  
1  750mL       7 $36.99  2016-12-31

In [12]:
endInv = endInv.rename(columns = {'onHand':'12_31_2016_Qty'})
endInv['InvCost12_31_2016'] = endInv['12_31_2016_Qty'] * endInv['Price']
endInv.head(2)

InventoryId  Store          City  Brand                  Description  \
0  1_HARDERSFIELD_58      1  HARDERSFIELD     58  Gekkeikan Black & Gold Sake   
1  1_HARDERSFIELD_62      1  HARDERSFIELD     62     Herradura Silver Tequila   

    Size  12_31_2016_Qty  Price     endDate  InvCost12_31_2016  
0  750mL              11 $12.99  2016-12-31            $142.89  
1  750mL               7 $36.99  2016-12-31            $258.93

In [13]:
endInv_Store = endInv[['Store','12_31_2016_Qty','InvCost12_31_2016']].groupby('Store').sum()
endInv_Store.head(2)

12_31_2016_Qty  InvCost12_31_2016
Store                                   
1               79827      $1,206,845.93
2               56671        $850,884.06

In [14]:
endInv_Store.sort_values(by='InvCost12_31_2016',ascending=False).head(10)

12_31_2016_Qty  InvCost12_31_2016
Store                                   
50             260717      $4,887,260.68
73             164589      $3,254,662.81
67             163765      $3,076,114.82
34             145829      $3,074,616.75
76             143866      $2,975,945.18
69             150848      $2,968,678.82
66             144579      $2,860,504.99
74             166015      $2,803,645.13
38             129397      $2,463,906.85
55             125584      $2,234,836.35

# Deliverable 2b

In [15]:
endInv_Brand = endInv[['Brand','12_31_2016_Qty','InvCost12_31_2016']].groupby('Brand').sum()
endInv_Brand.sort_values(by='InvCost12_31_2016',ascending=False).head(10)

12_31_2016_Qty  InvCost12_31_2016
Brand                                   
1233            15047        $526,494.53
3545            16770        $502,932.30
2753             7849        $470,861.51
8068            15608        $366,631.92
3405            12268        $355,649.32
4261            16769        $351,981.31
2757            11603        $336,370.97
2589             7922        $300,956.78
1376            13180        $276,648.20
2585            10487        $272,557.13

# Deliverable 3

In [16]:
Idqty_value_beg = begInv[['InventoryId','01_01_2016_Qty','InvCost01_01_2016']].groupby('InventoryId').sum()
Idqty_value_end = endInv[['InventoryId','12_31_2016_Qty','InvCost12_31_2016']].groupby('InventoryId').sum()
Change_qty_value = pd.merge(left = Idqty_value_beg, right = Idqty_value_end, how='outer', on=['InventoryId']).fillna(0)
Change_qty_value.head(2)

01_01_2016_Qty  InvCost01_01_2016  12_31_2016_Qty  \
InventoryId                                                          
10_HORNSEY_1000           $1.00             $14.99           $0.00   
10_HORNSEY_1001          $11.00             $65.89           $0.00   

                 InvCost12_31_2016  
InventoryId                         
10_HORNSEY_1000              $0.00  
10_HORNSEY_1001              $0.00

In [17]:
len(Change_qty_value)

256042

In [18]:
Change_qty_value = Change_qty_value.rename(columns = {'01_01_2016_Qty':'Beginning Quantity', 'InvCost01_01_2016':'Beginning Total $', 
                                                      '12_31_2016_Qty':'Ending Quantity', 'InvCost12_31_2016':'Ending Total $'})
Change_qty_value.head(10)

Beginning Quantity  Beginning Total $  Ending Quantity  \
InventoryId                                                                
10_HORNSEY_1000                $1.00             $14.99            $0.00   
10_HORNSEY_1001               $11.00             $65.89            $0.00   
10_HORNSEY_1005               $18.00            $629.82            $0.00   
10_HORNSEY_10058              $52.00            $779.48           $24.00   
10_HORNSEY_1006                $1.00              $3.99            $0.00   
10_HORNSEY_10062              $11.00             $98.89           $15.00   
10_HORNSEY_1009                $3.00            $149.97            $0.00   
10_HORNSEY_1012                $2.00             $69.98            $0.00   
10_HORNSEY_1013                $5.00             $94.95            $0.00   
10_HORNSEY_10164              $13.00            $207.87           $19.00   

                  Ending Total $  
InventoryId                       
10_HORNSEY_1000            $0.00  
10_HORNSEY_1001            $0.00  
10_HORNSEY_1005            $0.00  
10_HORNSEY_10058         $359.76  
10_HORNSEY_1006            $0.00  
10_HORNSEY_10062         $134.85  
10_HORNSEY_1009            $0.00  
10_HORNSEY_1012            $0.00  
10_HORNSEY_1013            $0.00  
10_HORNSEY_10164         $303.81

# Deliverable 4a

In [19]:
#The purchases file was to large for Jupter to run, it would freeze on the kernel and stop running
#Bescaue of that I just used the InvoicePurchases file for all of Deliverable 4
Purchases = pd.read_csv(r"C:\Users\msand\Oregon State University\Pesch, Heather Lynn - Bibitor_Data\2016Data\SimplifiedData\HW4 Sampled Data\InvoicePurchases12312016.csv")
Purchases.head(2)

VendorNumber                   VendorName InvoiceDate  PONumber  \
0           105  ALTAMAR BRANDS LLC           2016-01-04      8124   
1          4466  AMERICAN VINTAGE BEVERAGE    2016-01-07      8137   

       PODate     PayDate  Quantity  Dollars  Freight Approval  
0  2015-12-21  2016-02-16         6  $214.26    $3.47      NaN  
1  2015-12-22  2016-02-21        15  $140.55    $8.57      NaN

In [20]:
Purchases['InvoiceDate'] = pd.to_datetime(Purchases["InvoiceDate"])
Purchases["year"] = Purchases["InvoiceDate"].dt.year
Purchases.head(2)

VendorNumber                   VendorName InvoiceDate  PONumber  \
0           105  ALTAMAR BRANDS LLC           2016-01-04      8124   
1          4466  AMERICAN VINTAGE BEVERAGE    2016-01-07      8137   

       PODate     PayDate  Quantity  Dollars  Freight Approval  year  
0  2015-12-21  2016-02-16         6  $214.26    $3.47      NaN  2016  
1  2015-12-22  2016-02-21        15  $140.55    $8.57      NaN  2016

In [21]:
Purchases2016 = Purchases[Purchases["year"] == 2016]

In [22]:
Vendors2016 = Purchases2016[['VendorNumber','VendorName','Dollars']].groupby('VendorNumber').sum()
Vendors2016.sort_values(by='Dollars',ascending=False).head(10)

VendorName        Dollars
VendorNumber                                                                  
3960          DIAGEO NORTH AMERICA INC   DIAGEO NORTH AMERIC... $49,733,333.66
4425          MARTIGNETTI COMPANIES MARTIGNETTI COMPANIESMAR... $26,851,015.81
12546         JIM BEAM BRANDS COMPANY    JIM BEAM BRANDS COM... $23,539,441.79
17035         PERNOD RICARD USA          PERNOD RICARD USA  ... $23,460,507.21
480           BACARDI USA INC            BACARDI USA INC    ... $16,985,053.64
1392          CONSTELLATION BRANDS INC   CONSTELLATION BRAND... $15,187,741.32
1128          BROWN-FORMAN CORP          BROWN-FORMAN CORP  ... $13,139,921.12
9165          ULTRA BEVERAGE COMPANY LLP ULTRA BEVERAGE COMP... $12,433,070.97
3252          E & J GALLO WINERY         E & J GALLO WINERY ... $11,858,245.15
9552          M S WALKER INC             M S WALKER INC     ... $10,294,481.73

# Deliverable 4b

In [23]:
Vendors_Freight = Purchases2016[['VendorNumber','VendorName','Freight', "Quantity"]].groupby('VendorNumber').sum()
Vendors_Freight.sort_values(by='Freight',ascending=False).head(10)

VendorName     Freight  \
VendorNumber                                                                  
3960          DIAGEO NORTH AMERICA INC   DIAGEO NORTH AMERIC... $250,899.75   
4425          MARTIGNETTI COMPANIES MARTIGNETTI COMPANIESMAR... $139,976.94   
12546         JIM BEAM BRANDS COMPANY    JIM BEAM BRANDS COM... $120,230.57   
17035         PERNOD RICARD USA          PERNOD RICARD USA  ... $120,130.51   
480           BACARDI USA INC            BACARDI USA INC    ...  $86,153.58   
1392          CONSTELLATION BRANDS INC   CONSTELLATION BRAND...  $77,791.20   
1128          BROWN-FORMAN CORP          BROWN-FORMAN CORP  ...  $66,576.22   
9165          ULTRA BEVERAGE COMPANY LLP ULTRA BEVERAGE COMP...  $63,888.38   
3252          E & J GALLO WINERY         E & J GALLO WINERY ...  $59,594.41   
9552          M S WALKER INC             M S WALKER INC     ...  $52,199.20   

              Quantity  
VendorNumber            
3960           5329972  
4425           2547941  
12546          2673263  
17035          1599133  
480            1365360  
1392           2274715  
1128            973097  
9165           1017777  
3252           1791181  
9552           1305574

# Deliverable 4c

In [24]:
#limit 4a to values above 250,000
#left merge that new table with 4b
#new colum with freight/dollars
#sortby the new column decening and find tail and head

### NOTE

Because I used the InvoicePurchses file, my tables are going to look a little different than the ones shown in the check figures

In [25]:
High_Vendors = Vendors2016[Vendors2016["Dollars"] >= 250000]
High_Vendors_Freight = pd.merge(left = High_Vendors, right = Vendors_Freight, how="left", on=['VendorNumber', 'VendorName'])
High_Vendors_Freight.head(3)

VendorName  \
VendorNumber                                                      
480           BACARDI USA INC            BACARDI USA INC    ...   
516           BANFI PRODUCTS CORP        BANFI PRODUCTS CORP...   
653           STATE WINE & SPIRITS   STATE WINE & SPIRITS   ...   

                    Dollars    Freight  Quantity  
VendorNumber                                      
480          $16,985,053.64 $86,153.58   1365360  
516           $1,529,676.79  $8,034.30    213967  
653           $1,481,052.81  $7,757.25    149590

In [26]:
High_Vendors_Freight["Freight/$ %"] = (High_Vendors_Freight['Freight'] / High_Vendors_Freight['Dollars'])
High_Vendors_Freight["Freight/Unit"] = (High_Vendors_Freight['Freight'] / High_Vendors_Freight['Quantity'])
High_Vendors_Freight.head(2)

VendorName  \
VendorNumber                                                      
480           BACARDI USA INC            BACARDI USA INC    ...   
516           BANFI PRODUCTS CORP        BANFI PRODUCTS CORP...   

                    Dollars    Freight  Quantity  Freight/$ %  Freight/Unit  
VendorNumber                                                                 
480          $16,985,053.64 $86,153.58   1365360        $0.01         $0.06  
516           $1,529,676.79  $8,034.30    213967        $0.01         $0.04

In [27]:
High_Vendors_Freight['Freight/$ %'] = High_Vendors_Freight['Freight/$ %'].map('{:.2%}'.format)

In [28]:
High_Vendors_Freight.sort_values(by='Freight/$ %', ascending=False).head(10)

VendorName  \
VendorNumber                                                      
9625          WESTERN SPIRITS BEVERAGE COWESTERN SPIRITS BEV...   
516           BANFI PRODUCTS CORP        BANFI PRODUCTS CORP...   
17031         FLAG HILL WINERY & VINEYARDFLAG HILL WINERY & ...   
1590          DIAGEO CHATEAU ESTATE WINESDIAGEO CHATEAU ESTA...   
9744          FREDERICK WILDMAN & SONS   FREDERICK WILDMAN &...   
8892          TRINCHERO FAMILY ESTATES   TRINCHERO FAMILY ES...   
653           STATE WINE & SPIRITS   STATE WINE & SPIRITS   ...   
4425          MARTIGNETTI COMPANIES MARTIGNETTI COMPANIESMAR...   
8673          STE MICHELLE WINE ESTATES  STE MICHELLE WINE E...   
7153          PINE STATE TRADING CO      PINE STATE TRADING ...   

                    Dollars     Freight  Quantity Freight/$ %  Freight/Unit  
VendorNumber                                                                 
9625            $323,948.73   $1,746.69     52243       0.54%         $0.03  
516           $1,529,676.79   $8,034.30    213967       0.53%         $0.04  
17031           $281,173.13   $1,477.16     19188       0.53%         $0.08  
1590          $1,331,574.55   $7,083.48    183317       0.53%         $0.04  
9744            $736,944.55   $3,885.16     68087       0.53%         $0.06  
8892          $2,175,075.75  $11,225.72    340040       0.52%         $0.03  
653           $1,481,052.81   $7,757.25    149590       0.52%         $0.05  
4425         $26,851,015.81 $139,976.94   2547941       0.52%         $0.05  
8673          $3,007,382.16  $15,507.50    411345       0.52%         $0.04  
7153          $3,026,122.21  $15,390.66    410312       0.51%         $0.04

In [29]:
High_Vendors_Freight.sort_values(by='Freight/$ %', ascending=False).tail()

VendorName  \
VendorNumber                                                      
10000         MAJESTIC FINE WINES        MAJESTIC FINE WINES...   
1587          VINEYARD BRANDS INC        VINEYARD BRANDS INC...   
1485          CASTLE BRANDS CORP.        CASTLE BRANDS CORP....   
3960          DIAGEO NORTH AMERICA INC   DIAGEO NORTH AMERIC...   
6359          OLE SMOKY DISTILLERY LLC   OLE SMOKY DISTILLER...   

                    Dollars     Freight  Quantity Freight/$ %  Freight/Unit  
VendorNumber                                                                 
10000         $3,441,407.28  $17,293.91    438084       0.50%         $0.04  
1587          $1,182,486.91   $5,959.16    169264       0.50%         $0.04  
1485          $1,576,377.93   $7,956.22     98944       0.50%         $0.08  
3960         $49,733,333.66 $250,899.75   5329972       0.50%         $0.05  
6359            $377,754.50   $1,876.61    108698       0.50%         $0.02

# Deliverable 4d

In [30]:
ThouU = Purchases[Purchases['Quantity'] <= 1000]
HundF_ThouU = ThouU[ThouU['Freight'] > 100]
HundF_ThouU.head(2)

VendorNumber                   VendorName InvoiceDate  PONumber  \
11          1485  CASTLE BRANDS CORP.          2016-01-08      8152   
16          2242  DELICATO VINEYARDS INC       2016-01-06      8139   

        PODate     PayDate  Quantity   Dollars  Freight Approval  year  
11  2015-12-23  2016-02-19       320 $5,420.41  $179.26      NaN  2016  
16  2015-12-22  2016-02-10       808 $6,646.46  $127.05      NaN  2016

In [31]:
D4 = HundF_ThouU[['VendorNumber','VendorName','Freight', 'Quantity']].groupby('VendorNumber').sum()
D4.sort_values(by='Freight',ascending=False).head(10)

VendorName   Freight  \
VendorNumber                                                                
2561          EDRINGTON AMERICAS         EDRINGTON AMERICAS ... $2,944.67   
8664                                STOLI GROUP,(USA) LLC         $349.02   
7239                                REMY COINTREAU USA INC        $348.07   
653                                     STATE WINE & SPIRITS      $303.70   
6785                                PALM BAY INTERNATIONAL INC    $242.73   
10000                               MAJESTIC FINE WINES           $218.05   
3924                                HEAVEN HILL DISTILLERIES      $200.02   
8352                                LUXCO INC                     $196.61   
4692                                KOBRAND CORPORATION           $185.93   
1485                                CASTLE BRANDS CORP.           $179.26   

              Quantity  
VendorNumber            
2561             18120  
8664               975  
7239               382  
653                541  
6785               715  
10000              875  
3924               818  
8352               819  
4692               427  
1485               320

# Deliverable 4e

In [32]:
D5 = HundF_ThouU[['VendorNumber','VendorName','Freight','Dollars', 'Quantity']].groupby('VendorNumber').sum()

In [33]:
D5["Freight per dollar"] = (D5['Freight'] / D5['Dollars'])
D5["Freight per unit"] = (D5['Freight'] / D5['Quantity'])
D5['Freight per dollar'] = D5['Freight per dollar'].map('{:.2%}'.format)
D5.sort_values(by='Freight per dollar', ascending=False).head(10)

VendorName  Freight   Dollars  Quantity  \
VendorNumber                                                             
653               STATE WINE & SPIRITS     $303.70 $5,544.89       541   
7239          REMY COINTREAU USA INC       $348.07 $7,515.38       382   
6785          PALM BAY INTERNATIONAL INC   $242.73 $5,533.18       715   
2555          DISARONNO INTERNATIONAL LLC  $146.86 $3,506.41       385   
6213          MHW LTD                      $177.50 $4,821.83       412   
4692          KOBRAND CORPORATION          $185.93 $5,336.48       427   
1485          CASTLE BRANDS CORP.          $179.26 $5,420.41       320   
8352          LUXCO INC                    $196.61 $6,139.47       819   
10000         MAJESTIC FINE WINES          $218.05 $7,234.88       875   
3089          SIDNEY FRANK IMPORTING CO    $133.83 $4,443.66       379   

             Freight per dollar  Freight per unit  
VendorNumber                                       
653                       5.48%             $0.56  
7239                      4.63%             $0.91  
6785                      4.39%             $0.34  
2555                      4.19%             $0.38  
6213                      3.68%             $0.43  
4692                      3.48%             $0.44  
1485                      3.31%             $0.56  
8352                      3.20%             $0.24  
10000                     3.01%             $0.25  
3089                      3.01%             $0.35

In [34]:
D5.sort_values(by='Freight per dollar', ascending=False).tail(5)

VendorName   Freight  \
VendorNumber                                                                
2242                                DELICATO VINEYARDS INC        $127.05   
1273                                CALEDONIA SPIRITS INC         $109.80   
4848                                LAIRD & CO                    $100.37   
2561          EDRINGTON AMERICAS         EDRINGTON AMERICAS ... $2,944.67   
98450                               Serralles Usa LLC             $122.24   

                 Dollars  Quantity Freight per dollar  Freight per unit  
VendorNumber                                                             
2242           $6,646.46       808              1.91%             $0.16  
1273          $20,334.21       576              0.54%             $0.19  
4848          $18,937.82       982              0.53%             $0.10  
2561         $570,627.52     18120              0.52%             $0.16  
98450         $27,163.64       733              0.45%             $0.17

For Deliverable 4c, The top 10 largest vendors average a Freight per dollar (FD) of 0.53% and a Freight per unit (FU) of \\$0.05 while the bottom 5 average 0.50% and \\$0.05 respectively. In comparison, for Deliverable 4e the top 10 vendors average a FD of 3.84% and a FU of \\$0.45 with the bottom 5 coming in at 0.79% and $0.16 respectively.

For the top ten theres a difference of 3.31% for the FD and \\$0.40 for the FU. For the bottom 5 the difference is 0.29% and \\$0.11 respectively.

# Deliverable 5

In [35]:
#Which stores were the most profitable? Of those, what was their average inventory over the year? Could help look at inventory turnover. 

In [36]:
Sales = pd.read_csv(r"C:\Users\msand\Oregon State University\Pesch, Heather Lynn - Bibitor_Data\2016Data\SimplifiedData\HW4 Sampled Data\SalesFINAL12312016Sample.csv")
Sales.head()

InventoryId  Store  Brand                   Description  \
0   1_HARDERSFIELD_1009      1   1009       Rebel Yell Variety Pack   
1  1_HARDERSFIELD_10238      1  10238   Layer Cake Primitivo Puglia   
2  1_HARDERSFIELD_10239      1  10239       Cannonball Cab Svgn Cal   
3  1_HARDERSFIELD_10266      1  10266  Klinker Brick Old Vine Znfdl   
4   1_HARDERSFIELD_1029      1   1029  Fulton's Harvest Apple Pie L   

         Size  SalesQuantity  SalesDollars  SalesPrice   SalesDate  Volume  \
0  750mL 3 Pk              1        $49.99      $49.99  2016-01-02 $750.00   
1       750mL              2        $31.98      $15.99  2016-01-02 $750.00   
2       750mL              1        $13.99      $13.99  2016-01-02 $750.00   
3       750mL              1        $16.99      $16.99  2016-01-09 $750.00   
4       750mL              1         $9.99       $9.99  2016-01-03 $750.00   

   Classification  ExciseTax  VendorNo                   VendorName  
0               1      $0.79      8352  LUXCO INC                    
1               2      $0.22      4425        MARTIGNETTI COMPANIES  
2               2      $0.11      4425        MARTIGNETTI COMPANIES  
3               2      $0.11      9552  M S WALKER INC               
4               1      $0.79      3924  HEAVEN HILL DISTILLERIES

In [37]:
TopStores = Sales[['Store','SalesDollars']].groupby('Store').sum()
TopStores.sort_values(by='SalesDollars',ascending=False).head()

SalesDollars
Store              
76    $1,952,580.07
73    $1,726,571.73
34    $1,611,485.15
38    $1,490,789.19
66    $1,411,857.42

In [38]:
ABC = pd.merge(left=begInv_Store, right=endInv_Store, how="inner", on="Store")
ABC.head(2)

01_01_2016_Qty  InvCost01_01_2016  12_31_2016_Qty  InvCost12_31_2016
Store                                                                      
1               49917        $838,079.59           79827      $1,206,845.93
2               52925        $840,452.20           56671        $850,884.06

In [39]:
ABC["AverageInv"] = ((ABC['InvCost01_01_2016'] + ABC['InvCost12_31_2016'])/2)
ABC.head(2)

01_01_2016_Qty  InvCost01_01_2016  12_31_2016_Qty  InvCost12_31_2016  \
Store                                                                         
1               49917        $838,079.59           79827      $1,206,845.93   
2               52925        $840,452.20           56671        $850,884.06   

         AverageInv  
Store                
1     $1,022,462.76  
2       $845,668.13

In [40]:
mid = pd.merge(left=TopStores, right=ABC[["AverageInv"]], how="inner", on="Store")
mid.head(2)

SalesDollars    AverageInv
Store                            
1       $518,206.88 $1,022,462.76
2       $497,063.38   $845,668.13

In [41]:
mid["InvTurnover"] = (mid['SalesDollars']/mid["AverageInv"])
mid.head(2)

SalesDollars    AverageInv  InvTurnover
Store                                         
1       $518,206.88 $1,022,462.76        $0.51
2       $497,063.38   $845,668.13        $0.59

In [42]:
mid['InvTurnover'] = mid['InvTurnover'].map('{:.2%}'.format)
mid.head(3)

SalesDollars    AverageInv InvTurnover
Store                                        
1       $518,206.88 $1,022,462.76      50.68%
2       $497,063.38   $845,668.13      58.78%
3        $32,447.24   $128,957.70      25.16%

In [43]:
mid.sort_values(by='SalesDollars',ascending=False).head(10)

SalesDollars    AverageInv InvTurnover
Store                                        
76    $1,952,580.07 $2,964,181.81      65.87%
73    $1,726,571.73 $3,198,580.08      53.98%
34    $1,611,485.15 $3,182,893.50      50.63%
38    $1,490,789.19 $2,348,302.81      63.48%
66    $1,411,857.42 $2,916,769.44      48.40%
50    $1,319,851.61 $3,268,534.45      40.38%
67    $1,159,088.63 $3,077,846.72      37.66%
69    $1,145,068.89 $2,957,702.73      38.71%
60      $954,424.97 $1,374,947.43      69.42%
15      $811,485.28 $1,356,763.50      59.81%